app/models/repository_snapshot.rb

```diff
@@ -1,10 +1,13 @@
 # frozen_string_literal: true
 
 class RepositorySnapshot < RepositoryBase
+  enum status: { provisioning: 0, ready: 1, failed: 2 }
```

Comment:
I know it's not used anywhere ATM, but it would be nice to have an information that snapshot generation is in progress, I suggest we add a 'started' state. It would get set before the main transaction. It would help with potential debugging.

Reply:
It is called 'provisioning'